langchain/src/chat_models/anthropic.ts

```diff
@@ -8,19 +8,33 @@ import {
   BaseMessage,
   ChatGeneration,
   ChatResult,
-  MessageType,
 } from "../schema/index.js";
 import { getEnvironmentVariable } from "../util/env.js";
 import { BaseChatModel, BaseChatModelParams } from "./base.js";
 
-function getAnthropicPromptFromMessage(type: MessageType): string {
+function extractCustomRole(message: BaseMessage) {
+  if (!("role" in message && typeof message.role === "string")) {
```

Comment:
nit: Negation is confusing here, can we do `message.role !== undefined  && typeof message.role === "string"`?

Reply:
Same applies to below